render/src/html/mod.rs

```diff
@@ -0,0 +1,185 @@
+//! Defines the [`Html`] struct that is returned when rendering Unimarkup to HTML.
+
+use crate::render::OutputFormat;
+
+pub mod highlight;
+pub mod render;
+
+#[derive(Debug, Default)]
+pub struct HtmlAttribute {
+    pub name: String,
+    pub value: Option<String>,
+}
+
+#[derive(Debug, Default)]
+pub struct HtmlElement {
+    pub name: String,
+    pub attributes: HtmlAttributes,
+    pub content: Option<String>,
+}
+
+// Note: Both structs below are needed to implement `Display` for `Vec`.
+
+#[derive(Debug, Default)]
+pub struct HtmlAttributes(Vec<HtmlAttribute>);
+#[derive(Debug, Default)]
+pub struct HtmlElements(Vec<HtmlElement>);
+
+#[derive(Debug, Default)]
+pub struct Html {
+    pub head: HtmlElements,
+    pub body: HtmlElements,
```

Comment:
thanks. nice catch 👍

Just not sure about creating custom Head and Body structs.
They won't have attributes, because Unimarkup syntax does not allow it.
And `accidentally` passing something else won't be solved by this either, since both accept `HtmlElements`.

Reply:
I meant, accidentally passing some other `HtmlElement` where we meant `HtmlBody` for example. We can remove attributes, but I would suggest still creating separate elements with newtype pattern. 

It's really just a way to encode in type system that we want that specific element. In that case, compiler helps if implementation of `Display` is missing for `HtmlHead` for example.